torchgeo/datamodules/geo.py

```diff
@@ -0,0 +1,540 @@
+        """Implement one or more PyTorch DataLoaders for testing.
+
+        Returns:
+            A collection of data loaders specifying testing samples.
+
+        Raises:
+            MisconfigurationException: If :meth:`setup` does not define a
+                'test_dataset'.
+        """
+        dataset = self.test_dataset or self.dataset
+        if dataset is not None:
+            return DataLoader(
+                dataset=dataset,
+                batch_size=self.test_batch_size or self.batch_size,
+                shuffle=False,
+                num_workers=self.num_workers,
+            )
+        else:
+            msg = f"{self.__class__.__name__}.setup does not define a 'test_dataset'"
+            raise MisconfigurationException(msg)
+
+    def predict_dataloader(self) -> DataLoader[Dict[str, Tensor]]:
+        """Implement one or more PyTorch DataLoaders for prediction.
+
+        Returns:
+            A collection of data loaders specifying prediction samples.
+
+        Raises:
+            MisconfigurationException: If :meth:`setup` does not define a
+                'predict_dataset'.
+        """
+        dataset = self.predict_dataset or self.dataset
+        if dataset is not None:
+            return DataLoader(
+                dataset=dataset,
+                batch_size=self.predict_batch_size or self.batch_size,
+                shuffle=False,
+                num_workers=self.num_workers,
+            )
+        else:
+            msg = f"{self.__class__.__name__}.setup does not define a 'predict_dataset'"
+            raise MisconfigurationException(msg)
+
+    def on_after_batch_transfer(
+        self, batch: Dict[str, Tensor], dataloader_idx: int
+    ) -> Dict[str, Tensor]:
+        """Apply batch augmentations to the batch after it is transferred to the device.
+
+        Args:
+            batch: A batch of data that needs to be altered or augmented.
+            dataloader_idx: The index of the dataloader to which the batch belongs.
+
+        Returns:
+            A batch of data.
+        """
+        if self.trainer:
+            if self.trainer.training:
+                aug = self.train_aug or self.aug
+            elif self.trainer.validating or self.trainer.sanity_checking:
+                aug = self.val_aug or self.aug
+            elif self.trainer.testing:
+                aug = self.test_aug or self.aug
+            elif self.trainer.predicting:
+                aug = self.predict_aug or self.aug
+
+            batch = aug(batch)
+
+        return batch
+
+    def plot(self, *args: Any, **kwargs: Any) -> plt.Figure:
+        """Run the plot method of the dataset if one exists.
+
+        Args:
+            *args: Arguments passed to plot method.
+            **kwargs: Keyword arguments passed to plot method.
+
+        Returns:
+            A matplotlib Figure with the image, ground truth, and predictions.
+        """
+        dataset = self.val_dataset or self.dataset
```

Comment:
```diff
- dataset = self.val_dataset or self.dataset
+ dataset = self.dataset or self.val_dataset
```
self.val_dataset can be of type `torch.utils.data.Subset` which doesn't have a plot method and `or` will choose the first option.